### PR TITLE
Bugfix/map

### PR DESCRIPTION
**Pull request by Mathematou**
In this PR:

- Fixed the rotation of the areas
- Fixed the color of the remove button when editing the routes
- When a editing marker is clicked, blocked the event popup to show up
- Refreshed the list of profiles after modifying it.

### CODE REVIEW

File: app/src/main/java/com/github/sdpteam15/polyevents/model/map/GoogleMapVectorHelper.kt [hunk 1]
```diff
@@ -9,16 +9,18 @@ object GoogleMapVectorHelper {
...
-        val scalar = (vector.latitude * v.latitude + vector.longitude * v.longitude) / norm
-        return LatLng(scalar * v.latitude, scalar * v.longitude)
+    fun projectionVectorThroughCartesian(vector: LatLng, pos1: LatLng, pos2: LatLng): LatLng {
+        val vec = equirectangularProjection(vector, LatLng(0.0,0.0))
+        val v = equirectangularProjection(LatLngOperator.minus(pos1, pos2), LatLng(0.0,0.0))
+
+        val norm = v.first * v.first + v.second * v.second
+        val scalar = (vec.first * v.first + vec.second * v.second) / norm
+        return inverseEquirectangularProjection(Pair(scalar * v.first, scalar * v.second), LatLng(0.0,0.0))
     }
```
Comment (AlessioVerardo) [inline app/src/main/java/com/github/sdpteam15/polyevents/model/map/GoogleMapVectorHelper.kt:24]: Can we use the LatLng helper for this task ?

**Review by AlessioVerardo — APPROVED** [review]
Nice bug fix !

File: app/src/main/java/com/github/sdpteam15/polyevents/model/map/GoogleMapVectorHelper.kt (hunk 1 above)
Comment (Mathematou) [reply app/src/main/java/com/github/sdpteam15/polyevents/model/map/GoogleMapVectorHelper.kt:24]: No, there is a type missmatch. vec and v are of type Pair(Double, Double) and not LatLng.

**Review by Cell00phane — APPROVED** [review]

File: app/src/main/java/com/github/sdpteam15/polyevents/view/activity/EventActivity.kt [hunk 2]
```diff
@@ -78,6 +80,15 @@ class EventActivity : AppCompatActivity(), ReviewHasChanged {
...
     }
 
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        val id = item.itemId
+
+        if (id == android.R.id.home) {
+            finish()
+        }
+        return true
+    }
```
[the remark this answers is not in the document]
Comment (Mathematou) [reply app/src/main/java/com/github/sdpteam15/polyevents/view/activity/EventActivity.kt:90]: This is to have the correct behaviour when we press the back arrow. To go back to the activity with the same behaviour as with the back button of the phone.

**Pull request merged**